src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -81,16 +84,80 @@ protected virtual bool ProcessIndividualPackageJTokens(string filePath, ISingleF
         {
             var name = packageJToken["name"].ToString();
             var version = packageJToken["version"].ToString();
-
+            var authorToken = packageJToken["author"];
+            
             if (!SemanticVersion.TryParse(version, out _))
             {
                 Logger.LogWarning($"Unable to parse version \"{version}\" for package \"{name}\" found at path \"{filePath}\". This may indicate an invalid npm package component and it will not be registered.");
                 return false;
             }
 
-            var detectedComponent = new DetectedComponent(new NpmComponent(name, version));
+            var npmComponent = new NpmComponent(name, version);
```

Comment:
Can you update the class to have a constructor that takes in an optional NpmAuthor parameter? Having to create it and then do .Author on the next line seems unnecessary.

Reply:
Alternatively, for an optional parameter, something like

```C#
var npmComponent = new NpmComponent(name, version)
{
    Author = GetAuthor(authorToken, name, filePath),
};
```

But it's a style choice between those two options

Reply:
going with Cody's recommendation of introducing optional parameter in constructors, since I see that being used elsewhere in the code base.